Fix for Noetic (and two minor things)

tkazik opened the pull request:
- fixes message generation for Noetic
- gets rid of FindEigen.cmake deprecated warning (=> Eigen3)
- increases cmake min req version to get rid of cmake warnings

alexmillane reviewed (changes requested):
Thanks for this. Builds on Melodic for me.

However, we should sort out Jenkins before merging anything... more as a matter of principle than anything else...

I'll have a look this evening.

tkazik commented:
True...I just had a brief look:
- The handshake with ROVIO's lightweight_filtering on bitbucket fails
- Bitbucket has updated its cipher support last year (info [here](https://bitbucket.org/blog/update-to-supported-cipher-suites-in-bitbucket-cloud))
- Our trusty friend Jenkins still seems to be on 14.04 (EOL 2019 :)) and this is apparently the issue (old/no longer supported cipher)

Two options:
- Switch to ssh ([Link](https://stackoverflow.com/a/63717218/9189492))
- Update curl ([Link](https://stackoverflow.com/a/64403608/9189492))

Which one do you prefer?

PS: Same Jenkins issue probably also affects https://github.com/ethz-asl/ros_vrpn_client/pull/59 and https://github.com/ethz-asl/ros_vrpn_client/pull/58 (@rikba)

alexmillane commented:
Thanks the info @tkazik!

I see that all of our MAV jenkins jobs pull in all of our deps, of which lightweight_filtering (through rovio) is one. So I guess 100% of our buildjobs are broken right now. 

To confirm this I manually triggered a job for rovio (hasn't been built on jenkins in over a year) and sure enough it fails: https://jenkins.asl.ethz.ch/job/rovio/label=ubuntu/116/

I'm thinking that surely we have a 16.04 Jenkins slave? surely?! 

I'm clocking off for tonight though.

tkazik commented:
Yes, I think that currently all build jobs that include ROVIO **and** are executed by Jenkins on a 14.04/Trusty are broken. So there are some more options (besides the two above):

- Assign another/newer node for the build job (there are plenty of other/newer nodes, e.g. this [xenial](https://jenkins.asl.ethz.ch/computer/ibex-xenial/builds) or this [bionic](https://jenkins.asl.ethz.ch/computer/ibex-bionic/builds)). Especially the bionic node seems to have no jobs at all.
- We could remove the dependency on ROVIO for this package...but I guess it is in the deps for a reason, right?

I added the build status to the README, updated a few links and restructured it a bit...hope you don't mind.

michaelpantic commented:
we should probably kick out all builds for 14.04 (and 16.04?) eventually, not worth spending the time to maintain this stuff

alexmillane commented:
> we should probably kick out all builds for 14.04 (and 16.04?) eventually, not worth spending the time to maintain this stuff
> 
> (I estimate we have about 1 user still using them)

Yes but that user is probably the best-looking member of this team. 

... but yeah let's kick out 14.04.

michaelpantic commented:
> Yes but that user is probably the best-looking member of this team.

hahaha




![image](https://user-images.githubusercontent.com/1169724/104889535-a1c13300-596e-11eb-86d6-acfe1d54366b.png)

clanegge commented:
> Yes but that user is probably the best-looking member of this team.

Nah I am already on 18.04

tkazik commented:
lol guys, that escalated quickly...how about you ask some "mirror, mirror on the wall?"

Anyway, Jenkins seems to be willing to get some tender loving care (ssh or curl update on 14.04; or rather assign it to a newer node (preferable 18.04)). How do we proceed? I don't have the proper rights for the Jenkins machine...would be happy to look over someone's (virtual) shoulder.

michaelpantic commented:
> ssh or curl update on 14.04
hmm I think we shouldnt update this anymore.

I just checked, ros_vrpn_client is not set up as a multi-configuration jekins project, so it doesn't really care where it gets executed and thus may fail sometimes and not some other times.

Proposal: We set it up as a new multi-config project and assign it to right hosts. 
@Tim if you want we can to this together via zoom, e.g. later this week? Let me know by mail!
